File: 251229/정렬된 수 위치 알아내기/indices-of-sorted-array.py
n = int(input())
arr = list(map(int, input().split()))

sorted_arr = sorted(arr)

ans = []
for num in arr:
    for i in range(n):
        if sorted_arr[i] == num:
            idx = i+1
            while idx in ans:
                idx += 1
            ans.append(idx)
            break

print(*ans)

File: 251229/정렬된 수 위치 알아내기/indices-of-sorted-array.js
class Num {
    constructor(num, idx) {
        this.num = num;
        this.idx = idx;
    }
}

const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const arr = input[1].split(' ').map((v, i) => new Num(Number(v), i));

arr.sort((a, b) => {
    if (a.num !== b.num) return a.num - b.num;
    return a.idx - b.idx
});

const ans = Array(n).fill(0);
for (let i = 0; i < n; i++) {
    ans[arr[i].idx] = i+1;
}

ans.forEach(v => process.stdout.write(v + ' '));